### PR TITLE
Push extensions

### PR DESCRIPTION
**Pull request by mcweba**

### CODE REVIEW

File: gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        Iterator it = report.iterator();
+        while (it.hasNext()) {
+            ProcessingMessage msg = (ProcessingMessage) it.next();
+            JsonNode node = msg.asJson();
+            resultArray.add(new JsonObject(node.toString()));
+        }
+        return resultArray;
+    }
+
+    enum ValidationStatus {
```
Comment (floriankammermann) [inline gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java]: ValidationStatus already exists in org.swisspush.gateleen.validation, shouldn't we rather move this enum to gateleen-core.

File: gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        VALIDATED_POSITIV,VALIDATED_NEGATIV,COULD_NOT_VALIDATE
+    }
+
+    class ValidationResult {
```
Comment (floriankammermann) [inline gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java]: ValidationResultalready exists in org.swisspush.gateleen.validation, shouldn't we rather move this class to gateleen-core.

File: gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceManager.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    public List<ConfigurationResourceObserver> getObserversByResourceUri(String resourceUri) {
+        List<ConfigurationResourceObserver> resourceObservers = getObservers().get(resourceUri);
+        if (resourceObservers == null) {
+            resourceObservers = new ArrayList<>();
+        }
+        return resourceObservers;
+    }
+
+    private enum ConfigurationResourceChangeType {
+        CHANGE, RESET;
```
Comment (floriankammermann) [inline gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceManager.java]: Is reset the right wording? As I understand the resource is getting deleted and the Observers have to informed about that. Perhaps just better documentation needed.

File: gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java (hunk 1 above)
Comment (mcweba) [inline gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java]: done.

File: gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java (hunk 2 above)
Comment (mcweba) [inline gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceValidator.java]: done.

File: gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceManager.java (hunk 3 above)
Comment (mcweba) [inline gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceManager.java]: changed wording from "reset" to "remove" and enhanced documentation

**Pull request merged**